pkg/engine/jmespath/functions_test.go

```diff
@@ -391,63 +391,395 @@ func Test_LabelMatch(t *testing.T) {
 }
 
 func Test_Add(t *testing.T) {
-	jp, err := New("add(`12`, `13`)")
-	assert.NilError(t, err)
+	testCases := []struct {
+		test           string
+		expectedResult interface{}
+		err            bool
+		retFloat       bool
+	}{
+		{
+			test: "add('12', '13s')",
+			err:  true,
+		},
+		{
+			test: "add('12Ki', '13s')",
+			err:  true,
+		},
+		{
+			test: "add('12s', '13')",
+			err:  true,
+		},
+		{
+			test: "add('12s', '13Ki')",
+			err:  true,
+		},
+		{
+			test:           "add(`12`, `13`)",
+			expectedResult: 25.0,
+			retFloat:       true,
+		},
+		{
+			test:           "add(`12`, '13s')",
+			expectedResult: `25s`,
+		},
+		{
+			test:           "add('12s', '13s')",
+			expectedResult: `25s`,
```

Comment:
What will be the result of add('12.3s', '45.6s')? Can we take decimals for duration?

Reply:
Yes, we can, the result would be `57.9s` and if say we have `add('12.3s', '55.6s')` it'd be `1m7.9s`.